### PR TITLE
James 1799 - Port ES test suite to Scanning Search and Lucene

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            new MessageParser());
+        storeMailboxManager.setMessageSearchIndex(listeningMessageSearchIndex);
+        storeMailboxManager.init();
+    }
+
+    /**
+     * 15 tests out of 54 are failing
+     */
+
+    @Override
```
Comment (rouazana) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java]: I think @Ignoring them too is more eloquent.

File: mailbox/scanning-search/pom.xml [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                    <groupId>${project.groupId}</groupId>
+                    <artifactId>apache-james-mailbox-api</artifactId>
+                </dependency>
+                <dependency>
+                    <groupId>org.apache.james</groupId>
+                    <artifactId>apache-james-mailbox-store</artifactId>
+                </dependency>
+                <dependency>
+                    <groupId>org.apache.james</groupId>
+                    <artifactId>apache-james-mailbox-memory</artifactId>
```
Comment (rouazana) [inline mailbox/scanning-search/pom.xml]: order ^^

File: mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            new MessageParser());
+        storeMailboxManager.setMessageSearchIndex(listeningMessageSearchIndex);
+        storeMailboxManager.init();
+    }
+
+    /**
+     * 32 tests out of 54 are failing
+     */
+
+    @Override
```
Comment (rouazana) [inline mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java]: idem, you can @Ignore these tests.

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java (hunk 1 above)
Comment (aduprat) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java]: Right

File: mailbox/scanning-search/pom.xml (hunk 2 above)
Comment (aduprat) [inline mailbox/scanning-search/pom.xml]: OK

File: mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java (hunk 3 above)
Comment (aduprat) [inline mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java]: OK

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import org.apache.james.mailbox.inmemory.InMemoryMailboxManager;
+import org.apache.james.mailbox.inmemory.InMemoryMailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.JVMMailboxPathLocker;
+import org.apache.james.mailbox.store.MailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.MockAuthenticator;
+import org.apache.james.mailbox.store.mail.model.impl.MessageParser;
+import org.apache.james.mailbox.store.search.AbstractListeningMessageSearchIndexTest;
+import org.apache.lucene.store.RAMDirectory;
+
+public class LuceneMessageSearchIndexTest extends AbstractListeningMessageSearchIndexTest {
```
Comment (mbaechler) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java]: how many tests actually work ?

File: mailbox/scanning-search/src/main/java/org/apache/james/mailbox/store/search/SimpleListeningMessageSearchIndex.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.SearchQuery;
+import org.apache.james.mailbox.model.UpdatedFlags;
+import org.apache.james.mailbox.store.mail.MessageMapperFactory;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mailbox.store.mail.model.MailboxMessage;
+
+public class SimpleListeningMessageSearchIndex extends ListeningMessageSearchIndex {
```
Comment (mbaechler) [inline mailbox/scanning-search/src/main/java/org/apache/james/mailbox/store/search/SimpleListeningMessageSearchIndex.java]: I don't get it, why do we implement a listener if we don't care to listen at all ?

File: mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+import org.apache.james.mailbox.acl.UnionMailboxACLResolver;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.inmemory.InMemoryMailboxManager;
+import org.apache.james.mailbox.inmemory.InMemoryMailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.JVMMailboxPathLocker;
+import org.apache.james.mailbox.store.MailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.MockAuthenticator;
+import org.apache.james.mailbox.store.mail.model.impl.MessageParser;
+
+public class SimpleMessageSearchIndexTest extends AbstractListeningMessageSearchIndexTest {
```
Comment (mbaechler) [inline mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java]: how many tests working ?

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java [hunk 7]
```diff
@@ -56,63 +57,63 @@ protected void initializeMailboxManager() throws Exception {
      * 15 tests out of 54 are failing
      */
 
-    @Override
+    @Ignore
```
Comment (mbaechler) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java]: you should keep @Override if you want the IDE to help you on method rename

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java (hunk 4 above)
Comment (aduprat) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java]: Line 56:
```
    /**
     * 15 tests out of 54 are failing
     */
```

File: mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java (hunk 6 above)
Comment (aduprat) [inline mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java]: Line 54:
```
    /**
     * 32 tests out of 54 are failing
     */
```

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java (hunk 7 above)
Comment (aduprat) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java]: Ok

File: mailbox/scanning-search/src/main/java/org/apache/james/mailbox/store/search/SimpleListeningMessageSearchIndex.java (hunk 5 above)
Comment (aduprat) [inline mailbox/scanning-search/src/main/java/org/apache/james/mailbox/store/search/SimpleListeningMessageSearchIndex.java]: Ok

**Comment by aduprat** [comment]
test this please

File: mailbox/scanning-search/pom.xml [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+                    <version>${assertj-3.version}</version>
+                    <scope>test</scope>
+                </dependency>
+                <dependency>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-api</artifactId>
+                </dependency>
+                <dependency>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-simple</artifactId>
```
Comment (mbaechler) [inline mailbox/scanning-search/pom.xml]: bad idea, use logback for that

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractListeningMessageSearchIndexTest.java [hunk 9]
```diff
@@ -0,0 +1,655 @@
...
+import org.apache.james.mailbox.store.StoreMessageManager;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+
+public abstract class AbstractListeningMessageSearchIndexTest {
```
Comment (mbaechler) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractListeningMessageSearchIndexTest.java]: you should remove Listening from the class name

File: mailbox/scanning-search/pom.xml (hunk 8 above)
Comment (aduprat) [inline mailbox/scanning-search/pom.xml]: OK

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractListeningMessageSearchIndexTest.java (hunk 9 above)
Comment (aduprat) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractListeningMessageSearchIndexTest.java]: OK

**Comment by mbaechler** [comment]
:+1:

**Comment by rouazana** [comment]
:+1:

**Pull request merged**